modules/datastore/modules/datastore_mysql_import/src/Service/MysqlImport.php

```diff
@@ -153,16 +159,22 @@ protected function getFirstLineFromFile(string $file_path): string {
     if (!isset($f) || $f === FALSE) {
       throw new FileException(sprintf('Failed to open resource file "%s".', $file_path));
     }
-    // Attempt to retrieve the first line from the resource file.
-    $header_line = fgets($f);
-    // Close the resource file since it is no longer necessary.
+
+    // Attempt to retrieve the columns from the resource file.
+    $columns = fgetcsv($f);
+    // Attempt to read the column lines from the resource file.
+    $end_pointer = ftell($f);
+    rewind($f);
+    $column_lines = fread($f, $end_pointer);
+
+    // Close the resource file, since it is no longer needed.
     fclose($f);
-    // Ensure the first line of the resource file was successfully read.
-    if (!isset($header_line) || $header_line === FALSE) {
-      throw new FileException(sprintf('Failed to read header line from resource file "%s".', $file_path));
+    // Ensure the columns of the resource file were successfully read.
+    if (!isset($columns) || $columns === FALSE) {
+      throw new FileException(sprintf('Failed to read columns from resource file "%s".', $file_path));
     }
 
-    return $header_line;
+    return ['columns' => $columns, 'lines' => $column_lines];
```

Comment:
```suggestion
    return [$columns, $column_lines];
```
Can we simplify this? Maybe this is a common construction but I think it would be a bit clearer, if we have to return a list, to just use a simple array,. that's what i'm more used to seeing. (Not going to die on this hill though if you feel strongly this is better)

Reply:
I can see the benefit of it being either way. I prefer the verbosity, but if it's bothering you, I don't mind to change it.